### ISSUE TITLE
Support static member discovery in default memberdata data discoverer

### ISSUE THREAD

**Issue by boerdereinar** [xunit/xunit#2799]
The default data discover currently only supports test discovery in classes and their base types; however, interfaces can also contain static members which can contain test data.

```cs
public abstract class Repro<TSelf> where TSelf : ITest
{
	[Theory]
	[MemberData(nameof(TSelf.TestData))]
	public void Test(int value)
	{
	}
}

public sealed class ReproImpl : Repro<ReproImpl>, ITest
{
}

public interface ITest
{
	static virtual TheoryData<int> TestData => new() { 1, 2, 3 };
}
```

**Comment by bradwilson** [comment]
This will also likely need to include an update to the analyzers to recognize this new pattern (version limited by whatever version we choose to ship this feature into).

**Comment by JamesTerwilliger** [comment]
@bradwilson I can poke at this one some more and add tests, but I gave a first try on this one here: https://github.com/xunit/xunit/pull/2861.

**Comment by bradwilson** [comment]
Available in `0.1.1-pre.372` https://xunit.net/docs/using-ci-builds
